ssd1306/ssd1306.go

```diff
@@ -234,16 +260,47 @@ func (d *Device) SetPixel(x int16, y int16, c color.RGBA) {
 }
 
 // GetPixel returns if the specified pixel is on (true) or off (false)
-func (d *Device) GetPixel(x int16, y int16) bool {
+func (d *DeviceOf[T]) GetPixel(x int16, y int16) bool {
 	if x < 0 || x >= d.width || y < 0 || y >= d.height {
 		return false
 	}
 	byteIndex := x + (y/8)*d.width
 	return (d.buffer[byteIndex] >> uint8(y%8) & 0x1) == 1
 }
 
+var errOutOfBounds = errors.New("out of bounds")
+
+// DrawBitmap copies the bitmap to the internal buffer on the screen at the
+// given coordinates. It returns once the image data has been sent completely.
+func (d *DeviceOf[T]) DrawBitmap(x, y int16, bitmap pixel.Image[T]) error {
+	w, h := bitmap.Size()
+
+	k, i := d.Size()
+	if x < 0 || y < 0 || w <= 0 || h <= 0 ||
+		x >= k || (x+int16(w)) > k || y >= i || (y+int16(h)) > i {
+		return errOutOfBounds
+	}
+
+	// set area to draw
+	d.Command(COLUMNADDR)
+	d.Command(d.resetCol[0] + uint8(x))
+	d.Command(d.resetCol[0] + uint8(x) + uint8(w) - 1)
+
+	// TODO: handle when the bitmap is not aligned to 8 vertical pixels
+	d.Command(PAGEADDR)
+	d.Command((d.resetPage[0] + uint8(y)) / 8)
+	d.Command((d.resetPage[0] + uint8(y) + uint8(h)) / 8)
+
+	d.Tx(bitmap.RawBuffer(), false)
+	return nil
```

Comment:
For a prototype it may be easier to go through the bitmap and set it manually:

```go
width, height := bitmap.Size()
for by := 0; by < height; by++ {
    for bx := 0; bx < width; bx++ {
        d.SetPixel(x+bx, y+by, bitmap.Get(bx, by).RGBA())
    }
}
```

It'll be slower than necessary but it has a good chance of actually working.